src/internal/storage/fileset/reader.go

```diff
@@ -74,7 +74,7 @@ func (r *Reader) Shards(ctx context.Context, opts ...index.Option) (pathRanges [
 	ctx = pctx.Child(ctx, "", pctx.WithFields(LogIndex(prim.Additive, "startIdx")))
 	ir := index.NewReader(r.chunks, nil, prim.Additive, opts...)
 	pathRanges, err = ir.Shards(ctx)
-	return pathRanges, nil
+	return pathRanges, err
```

Comment:
This could just be `return ir.Shards(ctx)`, so long as errcheck is happy. Otherwise, you’ll need to wrap the error.

Reply:
True! I chose to wrap the error instead though